sources/packages/api/src/route-controllers/institution-locations/institution-locations.controller.ts

```diff
@@ -94,6 +77,48 @@ export class InstitutionLocationsController extends BaseController {
     return createdInstitutionlocation.id;
   }
 
+  @Patch(":locationId")
+  async update(
+    @Param("locationId") locationId: number,
+    @Body() payload: InstitutionLocationTypeDto,
+    @UserToken() userToken: IUserToken
+    ): Promise<number> {
+      // Validate the location data that will be saved to SIMS DB.
+      const dryRunSubmissionResult = await this.formService.dryRunSubmission(
+        "institutionlocationupdation",
+        payload,
+      );
+  
+      if (!dryRunSubmissionResult.valid) {
+        throw new UnprocessableEntityException(
+          "Not able to update the institution location due to an invalid request.",
+        );
+      }
+      
+      //To retrive institution id
+      const institutionDetails = await this.institutionService.getInstituteByUserName(
```

Comment:
This method needs to validate if the user updating the location has access to it.